### DISCUSSION TITLE
Wrong value for CalculatedAmount Version 14.1.0

### DISCUSSION THREAD

**Discussion by rimmel-ai** [stephanstapel/ZUGFeRD-csharp#415] [General]
For Austria I tested an invoice with 0,20,10,13 VAT. 
The value for "CalculatedAmount" 11.51 is wrong it should be 11.50

                    desc.AddApplicableTradeTax(basisAmount: 88.50m,
                                       percent: 13.00m,
                                       typeCode: TaxTypes.VAT,
                                       categoryCode: TaxCategoryCodes.S,
                                       allowanceChargeBasisAmount: 0.00m
                                      );

Result:

     <ram:ApplicableTradeTax>
        <ram:CalculatedAmount>11.51</ram:CalculatedAmount>
        <ram:TypeCode>VAT</ram:TypeCode>
        <ram:BasisAmount>88.50</ram:BasisAmount>
        <ram:CategoryCode>S</ram:CategoryCode>
        <ram:RateApplicablePercent>13.00</ram:RateApplicablePercent>
      </ram:ApplicableTradeTax>

Any idea?

**Comment by HenriSHS** [comment]
I took a look in the documentation and from my understanding 11.51 is the right CalculatedAmount.

> 7.1.8 Rounding rule in calculations
The rules that apply to calculate a Factur-X require a rounding calculation at certain stages (e.g. when
multiplying or dividing). The rounding method is that of the nearest value, with the rule for determining the
residual fraction to 0.5 as follows:
• For positive numbers: rounded up. Example: 13.455 rounded up to 2 digits gives 13.46

[https://github.com/stephanstapel/ZUGFeRD-csharp/blob/master/documentation/zugferd23en/Documentation/1.%20FACTUR-X%201.0.07%20EN.pdf](url) Page 35

**Comment by rimmel-ai** [comment]
Thanks.
The problem arises with gross amounts. Customers work with gross prices:
100/1.13=88.50
88.50*1.13=100.01=>VAT 11.51
So we have to solve it in the application.

**Comment by GambaJo** [comment]
I have the same problem. But why is this value calculated by this library? Should it not be just read out of the XML-file and not calculated? Even if it's wrong?
If I just want to visualize the XML for humans, now this library would deliver not the values out of the XML.

Example:
BasisAmout=221.49 and Percent=7

In the InvoiceDescriptor  TaxAmout=15.50
![grafik](https://github.com/user-attachments/assets/35bfb996-7618-4a53-ab6f-f090c4adb44d)

In the XML CalculatedAmout=15.51
![grafik](https://github.com/user-attachments/assets/521044aa-7b3d-42ec-9407-ccd5a9d43828)

At the moment it's not possible to get the original value from the XML (15.51) via the InvoiceDescriptor, so it's not possible to validate, if the values in the XML are correct.

Maybe @stephanstapel can tell us how the concept is.

**Comment by stephanstapel** [comment]
please update to the latest github version or wait for version 17.0
